src/containers/common/ProgramHomeContainer.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Actions } from 'jumpstate';
+import { Switch, Route, Redirect } from 'react-router-dom';
+
+import AstroHome from '../../components/astro/AstroHome';
+import DarienRoutesContainer from '../darien/DarienRoutesContainer';
+
+import {
+  PROGRAMS_INITIAL_STATE, PROGRAMS_PROPTYPES
+} from '../../ducks/programs';
+
+export class ProgramHomeContainer extends React.Component {
+  constructor() {
+    super();
+  }
+
+  componentDidMount() {
+    if (!this.props.selectedProgram) {
+      Actions.getProgram({ programs: this.props.programs, param: this.props.match.params.program });
+    }
+  }
+
+  componentWillUnmount() {
+    Actions.programs.selectProgram(PROGRAMS_INITIAL_STATE.program);
+  }
+
+  render() {
+    // How do I pass props down consistently? They only get initial props, not the updated ones.
```

Comment:
I accidentally asked this in PR #57 instead of this PR, but essentially: which updated props were you trying to pass down, but this component didn't register?

Reply:
I was trying to pass down props using the `Route` component:

`<Route {...props} path="/astro-101-with-galaxy-zoo/educators" component={AstroHome} />`

But what happened was that the child component at that route only got the default props, not the updated props once the selectedProgram was updated in the app store. I was hoping to be able to pass the prop to the child route like this so that I wouldn't have to connect both the container component and the child component to the store to get basically the same data. 

Either I don't understand how React Router v4 Route components work and this isn't supposed to work or it's a bug, but I didn't invest the time to figure that out and instead just have both the parent and child connected to the redux store. I left the comment to eventually come back to that and research in the docs and github issues to understand if this should work.